CHANGELOG.MD

```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v2.9.0] - 2025-12-10
```

Comment:
I'd probabyl merge these changes with 2.8.0 version which does not exist because it could not be built due to problems fixed in this PR